getinsomnia/insomnia
# feat: pre-request sdk object cleanups - INS-3379

### Please review this PR (https://github.com/Kong/insomnia/pull/7147) firstly as current one is based on it
### This one has been broken down to following smaller PRs:
https://github.com/Kong/insomnia/pull/7170
https://github.com/Kong/insomnia/pull/7171
https://github.com/Kong/insomnia/pull/7172
https://github.com/Kong/insomnia/pull/7173
https://github.com/Kong/insomnia/pull/7174

Changes:
- [x] enabled manipulation certificate and proxy through `insomnia.request`
- [x] added a test for above
- [x] added a simple interpolator for `replaceIn` method
- [x] other cleanups
- [x] cover auth transforming in tests
- [x] add more modules to `require`
- [x] introduce `eval` for compatibility of existing usages
- [x] organize snippets into categories 
- [x] restart the hidden window if it is down

### How to test
Pls run this script and check outputs.
```
// eval the express and set it
const evalResult = eval('8+8');
insomnia.environment.set('evalResult', evalResult);

// require uuid and generate an id
const uuid = require('uuid');
insomnia.environment.set('uuid', uuid.NIL);

// print current proxy and modify it
console.log(insomnia.request.proxy.getProxyUrl());
insomnia.request.proxy.update({
        host: 'localhost',
        match: '<all_urls>',
        port: 8888,
        tunnel: false,
        authenticate: false,
        username: '',
        password: '',
});

// print current certificate and modify it
// print the original one
console.log('key:', insomnia.request.certificate.key.src);
console.log('cert:', insomnia.request.certificate.cert.src);
console.log('passphrass:', insomnia.request.certificate.passphrass);
console.log('pfx:', insomnia.request.certificate.pfx.src);
// update
insomnia.request.certificate.update({
    disabled: true,
    key: {src: 'invalid.key'},
    cert: {src: 'invalid.cert'},
    passphrase: '',
    pfx: {src: ''},
});

```
